helix-term/src/ui/editor.rs

```diff
@@ -696,6 +696,17 @@ impl EditorView {
         let key_result = self.keymaps.get_mut(&mode).unwrap().get(event);
         self.autoinfo = key_result.sticky.map(|node| node.infobox());
 
+        // `restore_indent` tag may be set by `commands::open`, we need to cancel it
+        // when we are not enter normal_mode command.
+        match &key_result.kind {
```

Comment:
There's a better way of fixing this without having to add fields to doc or modifying handle_keymap_event: We already track mode transitions as well as track how insert mode was entered and what keys were typed:

https://github.com/helix-editor/helix/blob/67bf4250caf14d3a632abdbfa367c04b318d782c/helix-term/src/ui/editor.rs#L1045-L1067

All you need to do is under the insert -> normal transition check if the command was open above/below (last insert.0) and if no keys were typed (last insert.1.is_empty?())

Reply:
Thanks for the tip :-D  I have make a little adjust